### PR TITLE
auth: implement feature flag for service account proxy

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

This PR will bypass the service account proxy if the feature flag FlagExternalServiceAccounts is disabled.

**Why do we need this feature?**

All new developments must be feature-flag protected.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/serviceaccounts/proxy/service.go [hunk 1]
```diff
@@ -19,70 +19,77 @@ import (
...
 
 	return s.proxiedService.AddServiceAccountToken(ctx, serviceAccountID, cmd)
 }
 
 func (s *ServiceAccountsProxy) CreateServiceAccount(ctx context.Context, orgID int64, saForm *serviceaccounts.CreateServiceAccountForm) (*serviceaccounts.ServiceAccountDTO, error) {
-	if !isNameValid(saForm.Name) {
-		s.log.Error("Unable to create service account with a protected name", "name", saForm.Name)
-		return nil, extsvcaccounts.ErrInvalidName
+	if s.isProxyEnabled {
+		if !isNameValid(saForm.Name) {
```
Comment (gamab) [inline pkg/services/serviceaccounts/proxy/service.go:57]: Nit.
```suggestion
	if s.isProxyEnabled && !isNameValid(saForm.Name) {
```

**Review by gamab — APPROVED** [review]
LGTM, good job! 
Please mind my small comments before merging :)

File: pkg/services/serviceaccounts/proxy/service.go (hunk 1 above)
Comment (linoman) [reply pkg/services/serviceaccounts/proxy/service.go:57]: I thought about this but decided against, this is code that will be removed after all.

File: pkg/services/serviceaccounts/proxy/service.go [hunk 2]
```diff
@@ -19,70 +19,77 @@ import (
...
-
 	return s.proxiedService.DeleteServiceAccount(ctx, orgID, serviceAccountID)
 }
 
 func (s *ServiceAccountsProxy) DeleteServiceAccountToken(ctx context.Context, orgID int64, serviceAccountID int64, tokenID int64) error {
-	sa, err := s.proxiedService.RetrieveServiceAccount(ctx, 0, serviceAccountID)
-	if err != nil {
-		return err
+	if s.isProxyEnabled {
+		sa, err := s.proxiedService.RetrieveServiceAccount(ctx, 0, serviceAccountID)
```
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/serviceaccounts/proxy/service.go:82]: 🤦

**Pull request merged**